File: src/backend/shared/event_types.py
"""Event type constants and data schemas for CloudEvents published to Event Grid."""

from pydantic import BaseModel, Field

# ---------------------------------------------------------------------------
# Event type constants
# ---------------------------------------------------------------------------

EVENT_SOURCE = "/integration-copilot/api"

EVENT_ARTIFACT_UPLOADED = "com.integration-copilot.artifact.uploaded.v1"
EVENT_ARTIFACT_SCAN_PASSED = "com.integration-copilot.artifact.scan-passed.v1"
EVENT_ARTIFACT_SCAN_FAILED = "com.integration-copilot.artifact.scan-failed.v1"
EVENT_ARTIFACT_PARSED = "com.integration-copilot.artifact.parsed.v1"
EVENT_ARTIFACT_PARSE_FAILED = "com.integration-copilot.artifact.parse-failed.v1"
EVENT_GRAPH_UPDATED = "com.integration-copilot.graph.updated.v1"
EVENT_GRAPH_BUILD_FAILED = "com.integration-copilot.graph.build-failed.v1"
EVENT_ANALYSIS_REQUESTED = "com.integration-copilot.analysis.requested.v1"
EVENT_ANALYSIS_COMPLETED = "com.integration-copilot.analysis.completed.v1"
EVENT_ANALYSIS_FAILED = "com.integration-copilot.analysis.failed.v1"

# ---------------------------------------------------------------------------
# Event data schemas
# ---------------------------------------------------------------------------


class ArtifactUploadedData(BaseModel):
    """Data payload for the ArtifactUploaded CloudEvent."""

    tenant_id: str = Field(alias="tenantId")
    project_id: str = Field(alias="projectId")
    artifact_id: str = Field(alias="artifactId")
    artifact_type: str = Field(alias="artifactType")
    blob_path: str = Field(alias="blobPath")
    file_size_bytes: int = Field(alias="fileSizeBytes")
    content_hash: str = Field(alias="contentHash")

    model_config = {"populate_by_name": True}

File: src/worker-parser-rust/src/event_types.rs
//! Event type constants for Integrisight.ai CloudEvents.
//!
//! Mirrors the Python `shared/event_types.py` module.

pub const EVENT_ARTIFACT_UPLOADED: &str = "com.integration-copilot.artifact.uploaded.v1";
pub const EVENT_ARTIFACT_SCAN_PASSED: &str = "com.integration-copilot.artifact.scan-passed.v1";
pub const EVENT_ARTIFACT_SCAN_FAILED: &str = "com.integration-copilot.artifact.scan-failed.v1";
pub const EVENT_ARTIFACT_PARSED: &str = "com.integration-copilot.artifact.parsed.v1";
pub const EVENT_ARTIFACT_PARSE_FAILED: &str = "com.integration-copilot.artifact.parse-failed.v1";
pub const EVENT_GRAPH_UPDATED: &str = "com.integration-copilot.graph.updated.v1";
pub const EVENT_GRAPH_BUILD_FAILED: &str = "com.integration-copilot.graph.build-failed.v1";
pub const EVENT_ANALYSIS_REQUESTED: &str = "com.integration-copilot.analysis.requested.v1";
pub const EVENT_ANALYSIS_COMPLETED: &str = "com.integration-copilot.analysis.completed.v1";
pub const EVENT_ANALYSIS_FAILED: &str = "com.integration-copilot.analysis.failed.v1";
